Tasks/AzureAppServiceManage/azureappservicemanage.ts

```diff
@@ -72,127 +74,145 @@ async function run() {
         var preserveVnet: boolean = tl.getBoolInput('PreserveVnet', false);
         var extensionList = tl.getInput('ExtensionsList', false);
         var extensionOutputVariables = tl.getInput('OutputVariable');
-        var subscriptionId = tl.getEndpointDataParameter(connectedServiceName, 'subscriptionid', true);
         var taskResult = true;
         var errorMessage: string = "";
         var updateDeploymentStatus: boolean = true;
 
-        var endPoint = await azureStackUtility.initializeAzureRMEndpointData(connectedServiceName);
+        var azureEndpoint: AzureEndpoint = await new AzureRMEndpoint(connectedServiceName).getEndpoint();
+        var resources: Array<any> = await new Resources(azureEndpoint).getResources('Microsoft.Web/Sites', webAppName);
 
-        if(slotName && slotName.toLowerCase() === 'production') {
-            specifySlotFlag = false;
-            slotName = null;
+        if(action != "Swap Slots" && !slotName) {
+            if(!resources || resources.length == 0) {
+                throw new Error(tl.loc('ResourceDoesntExist', webAppName));
+            }
+            else if(resources.length == 1) {
+                resourceGroupName = resources[0].id.split("/")[4];
+            }
+            else {
+                throw new Error(tl.loc('MultipleResourceGroupFoundForAppService', resourceGroupName));
+            }
         }
 
-        if(resourceGroupName === null) {
-            resourceGroupName = await azureRmUtil.getResourceGroupName(endPoint, webAppName);
-        }
+        tl.debug(`Resource Group: ${resourceGroupName}`);    
         switch(action) {
             case "Start Azure App Service": {
-                console.log(await azureRmUtil.startAppService(endPoint, resourceGroupName, webAppName, specifySlotFlag, slotName));
-                await waitForAppServiceToStart(endPoint, resourceGroupName, webAppName, specifySlotFlag, slotName);
+                var appService: AzureAppService = new AzureAppService(azureEndpoint, resourceGroupName, webAppName, slotName);
```

Comment:
can this be moved outside switch and reused in all cases ?

Reply:
This is not required for Swap slots, so made it duplicate.
Will check for other efficient ways,